mmv1/third_party/terraform/utils/config.go.erb

```diff
@@ -187,9 +185,13 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 
 	// Start DCL client instantiation
 	// TODO(slevenick): handle user agents
-	c.dclConfig = dcl.NewConfig(dcl.WithHTTPClient(client), dcl.WithUserAgent(c.userAgent), dcl.WithLogger(dclLogger{}))
-	c.clientDataprocDCL = dataprocDcl.NewClient(c.dclConfig)
-	c.clientEventarcDCL = eventarcDcl.NewClient(c.dclConfig)
+	dclClientOptions := dcl.WithHTTPClient(client)
+	dclUserAgentOptions:=dcl.WithUserAgent(c.userAgent)
+	dclLoggerOptions := dcl.WithLogger(dclLogger{})
+	// each product needs it own client currently since basepath can only be specified at
+	// the config level.
+	c.clientDataprocDCL = dataprocDcl.NewClient(dcl.NewConfig(dclClientOptions, dclUserAgentOptions, dclLoggerOptions))
```

Comment:
Should we use the base paths here?

Reply:
base paths aren't exposed by the dcl. They are [hardcoded](https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/services/google/eventarc/trigger_internal.go#L122)